librz/core/cmp.c

```diff
@@ -0,0 +1,361 @@
+// SPDX-FileCopyrightText: 2021 Dhruv Maroo <dhruvmaru007@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_cmp.h>
+#include <rz_asm.h>
+#include <rz_list.h>
+
+RZ_API RZ_OWN RzCompareData *rz_cmp_data_data(RZ_NONNULL RzCore *core, ut64 addr1, ut64 addr2, ut32 len) {
+	rz_return_val_if_fail(core, NULL);
+
```

Comment:
Well, I think we can drop those. But it will fail quicker if the boundaries are unmapped. Both approaches are fine, I think. If you prefer we could drop these checks completely.

Reply:
Yes I think we should just rely on the read operation and check failure there.